engine/apps/integrations/tests/test_views.py

```diff
@@ -459,3 +460,58 @@ def test_integration_grafana_endpoint_without_cache_has_alerts(
             call((alert_receive_channel.pk, data["alerts"][1]), kwargs={"received_at": now.isoformat()}),
         ]
     )
+
+
+@patch("apps.integrations.views.create_alert")
+@pytest.mark.parametrize(
+    "integration_type",
+    [
+        arc_type
+        for arc_type in AlertReceiveChannel.INTEGRATION_TYPES
+        if arc_type not in ["amazon_sns", "grafana", "alertmanager", "grafana_alerting", "maintenance"]
+    ],
+)
+@pytest.mark.django_db
+def test_integration_outdated_cached_model(
+    mock_create_alert,
+    make_organization_and_user,
+    make_alert_receive_channel,
+    integration_type,
+):
+    organization, user = make_organization_and_user()
+    alert_receive_channel = make_alert_receive_channel(
+        organization=organization,
+        author=user,
+        integration=integration_type,
+    )
+
+    # set an invalid cache value for the requested integration
+    cache_key = AlertChannelDefiningMixin.CACHE_KEY_SHORT_TERM + "_" + alert_receive_channel.token
+    cache.set(cache_key, '{"some": "invalid json model"}')
```

Comment:
just out of curiosity, do you know under what scenarios would our cache contain invalid JSON for an integration?

Reply:
It happened earlier today, during the rollout, in which we introduced changes to the `AlertReceiveChannel` model, raising some [exceptions](https://ops.grafana-ops.net/explore?schemaVersion=1&panes=%7B%22hum%22:%7B%22datasource%22:%22c-R8UWvVk%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22amixr-prod%5C%22,%20job%3D%5C%22amixr-prod%2Famixr-integrations%5C%22%7D%20%7C%3D%20%5C%22django.core.serializers.base.DeserializationError%5C%22%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22c-R8UWvVk%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%221706023840486%22,%22to%22:%221706024722486%22%7D%7D%7D&orgId=1) then when trying to deserialize the cached version (which was pre- db migration)